src/ArrayManager.inl

```
@@ -166,6 +180,13 @@ void* ArrayManager::reallocate(void* pointer, size_t elems)
   auto space = CPU;
 #endif
 
+  for (int i = 0; i < NUM_EXECUTION_SPACES; i++) {
+    if(!pointer_record->m_owned[i]) {
+      CHAI_LOG("ArrayManager", "Cannot reallocate unowned pointer");
```

This is a warning, but we could throw an error here... @rfhaque @rhornung67